compiler/iface/IfaceType.hs

```diff
@@ -1203,6 +1203,11 @@ pprTyTcApp' ctxt_prec tc tys dflags style
   , rep `ifaceTyConHasKey` liftedRepDataConKey
   = kindType
 
+  | tc `ifaceTyConHasKey` funTyConKey
+  , IA_Vis (IfaceTyConApp rep IA_Nil) args <- tys
+  , rep `ifaceTyConHasKey` omegaDataConKey
+  = pprIfacePrefixApp ctxt_prec (parens arrow) (map (ppr_ty appPrec) (tys_wo_kinds_fun args))
```

Comment:
Do you know why this special casing needs to be?

Reply:
Without this code, many error messages that used to say `No instance for C ((->) a` say `No instance for C (FUN 'Omega a)`.